### DISCUSSION TITLE
Regarding apiman-deployment

### DISCUSSION THREAD

**Discussion by Devivaralakshmi** [apiman/apiman#2469] [Q&A]
Hi team, is there any references to deploy apiman on kubernetes cluster?

**Comment by msavy** [comment]
Hi @Devivaralakshmi.

I don't have any specific K8s guides yet. But I'm working on some stuff for my company. I know a lot of people already use Apiman on K8s with their own service templates.

If you want to create your own, then it should not be too difficult, and you can perhaps use the Docker Compose distro as a template?

Are there any K8s specific features you would like?

**Reply by Gautham-anne** [reply]
hello is there any specific docker compose distro which is working?
i am trying with the 3.0.0 final distro which is causing some issues for me 
i also tried the apiman/0n-wildfly image which caused the invalid redirect url issue 
i tried solving it by changing the apiman-realm-for-keycloak.json files but it didnt solve it

**Reply by msavy** [reply]
It should be working. If it is not, please open an issue ticket so I can look into it.

`apiman/on-wildfly` is deprecated because we can't bundle Keycloak any more (due to changes to Keycloak).

**Reply by Gautham-anne** [reply]
can you give a docker image which is compatible with k8 deployment?
i am currently deploying as microservices with separate pods for keycloak, gateway, manager, and reverse proxy

**Reply by Gautham-anne** [reply]
A rough walkthrough would also be helpful

**Reply by msavy** [reply]
There is no Docker image that is explicitly created with K8s in mind, but it's very easy to get it running in K8s and I know a bunch of people who have done it successfully. What you describe is perfectly reasonable, although you can also expose the gateway directly via load balancer/ingress rather than reverse proxy — both perfectly valid, but everyone's deployments are a bit different depending on their system architecture and so forth. 

For example, it's very easy to configure Apiman using environment variables and/or system properties. That pattern works well with K8s.

I am currently working on Apiman K8s-specific stuff like operators, templates, and extremely high-performance distributed rate limiting via my company [Black Parrot Labs Enterprise offering](https://www.blackparrotlabs.io), rather than upstream (in order to try to survive financially so that I can continue maintaining and developing the project).

Of course, I am willing to take feature requests too.

**Reply by Gautham-anne** [reply]
can you refer me someone who has done it?
ill check with them if there is something which i am doing wrong?

**Reply by msavy** [reply]
Sorry, I have NDAs with the companies I've worked with in the past that mean I'm not allowed to disclose their information or code samples.

If any community members want to share the specifics you are asking for, they are most welcome. 

And of course, any company is welcome to reach out to me/BPL, as that supports the project financially 🙏.

**Reply by Gautham-anne** [reply]
ok is it possible to schedule a support call so we can see if there is anything wrong with my deployment?

**Reply by msavy** [reply]
I replied to your enquiry, but didn't hear back, so letting you know here also in case my response was spam-binned. If you have changed your mind, that's also fine; please let me know so I don't invest any more time into it.

**Reply by msavy** [reply]
Just to close off the thread — unfortunately this person was wasting my time.
